src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -101,6 +102,33 @@ def _get_attr(self, data, attr, empty_value=None):
         return empty_value
 
 
+class DashboardWidgetMetricsQuerySerializer(CamelSnakeSerializer):
+
+    id = serializers.CharField(required=False)
+    fields = serializers.ListField(child=serializers.CharField(), required=False)
+    name = serializers.CharField(required=False, allow_blank=True)
+    conditions = serializers.CharField(required=False, allow_blank=True)
+    groupby = serializers.ListField(required=False)
+    project_id = serializers.IntegerField(required=False)
+
+    required_for_create = {"fields", "conditions", "project_id"}
+
+    validate_id = validate_id
+
+    def validate(self, data):
+        if not data.get("id"):
+            keys = set(data.keys())
+            if self.required_for_create - keys:
```

Comment:
Should we validate the fields + groupby etc here too?

Reply:
yes. I agree that it would be nice to validate the fields